### DISCUSSION TITLE
Temperature sensors, especially extreme frosting on external unit

### DISCUSSION THREAD

**Discussion by ando2040** [revk/xESP32-Faikout#194] [Q&A]
> Hi all,
> 
> I've just installed 5 Faikin boards into my Daikin split system. It's working great and providing tremendous control and (sometimes too much) data. For reference, FTXR18WVJUW9 & FTXR09WVJUW9 Emura series, and 3x FTXS09WVJU serviced by a 5MXS48WVJU9 outdoor unit.
> 
> Quick question, is it possible for individual energy consumption of the outdoor and interior units to be exposed to MQTT? It would be great to be able to see what sort of usage they're using and add individual entities to Home Assistant.
> 
> Thanks again for a great project. It added in seamlessly, was a fraction of the price Daikin wanted, and mostly importantly completely local.
> 
> ![Image 2024-01-13 at 11 47](https://private-user-images.githubusercontent.com/49879127/296519817-74bb823f-8a59-4567-bea4-a90f92800d84.JPG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDYwMDMwNjUsIm5iZiI6MTcwNjAwMjc2NSwicGF0aCI6Ii80OTg3OTEyNy8yOTY1MTk4MTctNzRiYjgyM2YtOGE1OS00NTY3LWJlYTQtYTkwZjkyODAwZDg0LkpQRz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIzVDA5MzkyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyNDBmYTRkMjJkOTc1ODQzYTZjMmNjYTg3ZGZiZTNjNzhmYjJlM2M5M2IyMzQyNzY2MmMzZDNiMzdjMjBkOGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.i9W3rc52RVx5US69EzFRfIAuboHxZdwZBD1v0OoENw0)

@Rev-777 I started a new discussion to avoid dragging the original away, but apologies I am slightly off topic for the whole forum here. I've bought one Faikin module to test but not screwed up the courage to install yet - I'm getting very close due to frustration with the temperature readouts from the built in sensors. 

Our installation is a 5MXM90A2V1B9 with 3xFTXM25R2V1B, 1xFTXM20R2V1B and 1xFTXM60R2V1B

There is a second (possibly related) issue with frost/defrost cycles on the 5MXM at sub-zero temperatures. Frost builds up, and when it defrosts (up to hourly), heating is compromised for 15-20 minutes. Internal fans stop, the unit defrost takes 5 min (loads of water ejected) and then it takes 10-15 min to get back to full power. Have you noticed similar, since I see you are running in sub zero temp? 

I fear a similar issue with the external unit thermistor, as I am able to see with the internal units. I will know more if I get around to installing the Faikin (and potentially voiding my warranty - at present I'm trying to get Daikin to fix it but I'm aware this may be a futile exercise and workaround may be all I have.) 

Thanks in advance

**Comment by Rev-777** [comment]
Yes, the defrost cycle is completely normal and necessary. Go take a look at the outdoor unit’s fins just prior to defrost and it’ll be almost entirely covered in snow/frost/ice. The unit brings the hot liquid back to the condenser and melts this all away.

What have you been noticing incorrect?

**Reply by ando2040** [reply]
20 mins in every hour without heat just seems like a long time - are you saying that's about normal?

I agree prior to that period, the defrost is 100% necessary. The question is whether it should be so often and take so long.

**Reply by revk** [reply]
Get someone to check levels of coolant.

When creating the Faikin, making it log when it went in to "antifreeze" helped identify I had a problem. It was too little coolant in the system. Easily fixed by the installers.

**Reply by Rev-777** [reply]
Coolant is an excellent thought.

As a data point, my defrost cycle is typically 3-7 minutes long at the condenser, and the splits themselves off about 2 minutes past that before heat is produced again.

Hope that helps.

**Reply by ando2040** [reply]
Thanks - there was initially too little refrigerant - 2kg is precharged but a 5-way split with longish pipe runs needs more, the installer came back less than a week after first install, admitted the mistake and topped up to the max 3.3kg. This seemed to reduce the noise from the external unit but temp was above zero by then, so we didn't know until last week that it hadn't fixed the regular defrosts. He came back last night (again the cold snap is over for now) and checked the pressures in case of a leak - I understand all well on that front. 

I'm OK with the defrost taking 4-5 mins at the external unit - I can see what's happening and why it takes a while to drain - but the amount of times it does it (is once per hour within your experience?) and the amount of time to fully recover (yes, some heat comes back but it takes 15 min to fully get back up) are severely hampering the total output per hour! I don't think this is even the reason why max output is expected to be lower in cold temps, so this defrost time is on top? Our pump is 10kW nominal (@7degC) and 5kW minimum (@-15degC) so I reckoned on 7-8kW (@-5degC last week.) By the time the system was defrosting once per hour at its worst, it felt a lot less.

I think I will put the Faikin module I have on my 2kW unit (least warranty exposure) to get some data in the next cold snap as you suggest. It's frustrating that Daikin themselves give little to no window on what's happening outdoors, even though they could!

Edit: apologies if user 7C visits - accidental tag, now corrected to degC

**Reply by Rev-777** [reply]
Interesting! I don’t worry about warranty, to be honest. If I had an issue I’d simply remove Faikin and call my installer.

Faikin (in my mind) would be the last piece of hardware to create issues.

Zero issues so far; it’s nice gear.

**Reply by ando2040** [reply]
I expect I'll do the same once I've got over the hurdle of first install!

**Reply by ando2040** [reply]
First cold day since I fitted the Faikin and using it for troubleshooting what I think is extreme and regular frosting.

- Auto mode about 0720, refrigerant got to 37C at about 0730.
- By 0750, massive frosting and compressor working really hard. 
- Defrost at 0815. 
- Pattern repeated 0830-0910. 
- Then the air temp got higher and it didn't happen again - 1000 was a timed power off followed by manual power on.

Between 0720 and 0920 I only got about 50 minutes of refrigerant being above 37C.

This doesn't seem normal - or is it? My card YAML below if anyone would be kind enough to help me with theirs tomorrow (still cold in UK) please.

![image](https://github.com/revk/ESP32-Faikin/assets/153862115/18f3593b-6bdf-4be1-93cd-fa64c81bd9bf)

```
type: custom:apexcharts-card
header:
  show: true
  show_states: true
  colorize_states: true
graph_span: 4h
series:
  - entity: sensor.climate_study_comp
    name: Compressor
    yaxis_id: fan
    curve: stepline
    stroke_width: 2
    data_generator: ''
  - entity: sensor.climate_study_liquid
    name: Refrigerant
    yaxis_id: temp
    curve: smooth
    stroke_width: 2
    data_generator: ''
  - entity: sensor.study_b_outside_temperature
    name: Daikin external sensor
    yaxis_id: temp
    curve: stepline
    stroke_width: 2
    data_generator: ''
yaxis:
  - id: temp
  - id: fan
```
EDIT: the Faikin _outside sensor provides the same data as outside_temperature - I checked. I happen to have the daikin_residential integration installed, that does provide this sensor via Onecta (but not the compressor or liquid)

**Reply by revk** [reply]
Yeh too little refrigerant

**Reply by ando2040** [reply]
Installer will check pressure again next week - he phoned me straight away even though it's the weekend when he saw this chart (I'm trusting him not to shop me to Daikin and risk them voiding the warranty.) It should be maxed already (3.3kg for 5-way split as per manual.)

**Reply by ando2040** [reply]
Installer came back last night and added another 300g of R32 (the most he felt comfortable with, given it's already had the max recommended.)

This morning, maybe a small improvement (hard to tell as air temp was increasing). The unit maybe held on for longer without defrosting, but the refrigerant temperature arguably fell too far to be effective. I still don't feel like I know for sure how a 5-way split should behave in cold weather.

Here's the side of the unit just before defrost at 0745, air temp 2C so hardly extreme. 
![image](https://github.com/revk/ESP32-Faikin/assets/153862115/600b61dd-cadc-41bd-9b2d-7281fca19674)

The first half hour was 1x6kW, then a 2kW unit kicked in at 0600, then 3 x 2.5kW at 0630.  This is a Daikin-supported (and recommended by them) combination of 5 units. Obviously it will work the outdoor unit hard since it's oversubscribing it, but given it's supposed to be able to handle -15C should it be working this hard (and letting refrigerant temp drop to about 30C) at 1C? Wondering if there is a sensor issue.
![image](https://github.com/revk/ESP32-Faikin/assets/153862115/9dc2163e-528e-4910-973e-c97513c45894)

Any further thoughts appreciated. Evidence from other multi-splits in cold weather especially.

EDIT: After 8am it's just the 6kW unit programmed so back to comfortable for the compressor. Plus the higher air temp by then.

**Reply by ando2040** [reply]
I do have a warranty call into Daikin BTW. They'll probably turn up in a couple of months once the weather is nice and warm, but not warm enough I actually want cooling...

**Reply by chrisevans1001** [reply]
What was the outcome for you on this? We have a 5-way Daikin system using the same outdoor unit as you have but with a slight variance in indoor units. The heating has been great until we hit below 0 this week. Now the defrost cycles are regular and the indoor units are forcing cold air into the rooms that they've just heated. Not sure if this is expected or not. We're giving this method of heating a go for a while to see whether it is worth it as a permanent option and subsequently being able to take out the radiators.

**Reply by ando2040** [reply]
Daikin sent an engineer in June - this was not a priority.for them! Early days because I'm still only a couple of days into their suggested fix due to the return of cold weather, but I haven't seen any excessive frosting yet. 

I suspect like me you were using the obvious Auto mode. I have switched all my schedules to Heating mode at their suggestion and tentatively it's much better. 

The reasoning is to do with offsets. Auto uses a default 2C offset and it cannot be changed, so you get hot air at the top of the room where the sensor is, and the unit might try to switch to cooling (one symptom you described). Or, you try to set a high temperature in desperation and the external unit works too hard, causing the frosting (the other symptom.) Heating mode has a default 4.5C offset and it can be changed! 

Technically this *is documented* (not the frosting user-invoked consequence), on page 317 of the service manual! After you've changed everything to heating mode and monitored for a few days, if you need to fine tune a unit, start halfway down - not at the top of the page like the engineer did, or you will lose your cooling modes. I had to work that out for myself and reverse what he did. The factory default is not made explicit, only the offset from the factory offset! In other words you can fine-tune the offset 2C either side of the 4.5C default. If any of this works for you please do complain to Daikin that it's far too hidden! You shouldn't need to fine-tune, my tentative experience is the 4.5C default is OK. 

The temperature reported by onecta is the real sensor reading, not the corrected offset - it is expected to be wrong - which is IMHO sensible to engineers but a terrible user interface! I use BLE sensors at a sensible 1.2m from floor height as elsewhere on this repo, to monitor the perceived temperature. Switching all the units to faikins and connecting this external sensor would also have worked and I may yet still do that. But I was stubborn that Daikin as one of the biggest vendors in the world should be able to fix this in warranty...

![IMG-20240523-WA0007](https://github.com/user-attachments/assets/5e144928-3183-4557-a7cc-478df7ef7763)

**Reply by ando2040** [reply]
I'm at my PC now so I'll post a chart of the last 24h. Temperatures below 5C all this time, and our house is poorly insulated - that's a work in progress significantly (I believe and hope) dependent on a big window replacement project.

![image](https://github.com/user-attachments/assets/c573aa66-ee88-4d74-8814-0ee81d30a7fc)

The important thing to note is the delta between the traditional thermostat-height temperature (purple) and the unit-reported temperature (orange, converges back with perceived in the middle of the night when no heat is being applied.) This is what requires the offset, and the fixed 2C offset in Auto is demonstrably not enough. The unit is now capable of reaching the setpoint. It only actually did so at 3pm yesterday, but I believe that's because 10kW output shared between 5 units is a bit below what's required at our current level of insulation - we previously used a 20kW output gas boiler. At 1500 yesterday the observed offset was 4C though in fact I have altered the default for this unit to the maximum 6.5C offset (7-4) so I could probably dial it back to default.

Other notes on the graph: Auto 25C was a brief accident because the IR controller was still set for that - another poor user interface that confuses the heck out of my wife. 16C setpoint at 2045 was an experiment - successful in my view because nothing much happened, the unit kept running.

One other side benefit; the external unit does seem noticeably quieter in this heating only mode - I suspect the compressor isn't working as hard to counter the lowered efficiency due to frosting. If Daikin would listen, I would tell them it hangs on far too long before the inevitable defrost.
